### DISCUSSION TITLE
Render multiline go template

### DISCUSSION THREAD

**Discussion by kaldvelski** [helmfile/helmfile#617] [Questions]
When I use the `tpl` function to render a template file (via `readFile`) then I'm finding that only the first line of the file is being output.

I'm not sure if this is because `tpl` evaluates only strings and so the result of `readFile`, while a string, is that only the first line is evaluated. 

Is there a way to setup a more complex template using conditional statements etc. rather than writing it as a single pipeline?

BTW I want to use `.Release.Name` in the template file so need to evaluate as:
`{{``{{ tpl (readFile "templates/load.tpl") . }}``}}`

**Comment by yxxhero** [comment]
@kaldvelski please show some log or output. Thanks very much.

**Comment by kaldvelski** [comment]
Unfortunately there's not much I can show you because my whole problem is there is no output rendered from my .tpl file.

If the first line outputs content, or is unevaluated e.g.

`abc`
`{{ if something }}`
`{{ do this }}`
`{{end}}`

Then "abc" is the output

but if it's like this:

`{{ if something }}`
`{{ do this }}`
`{{end}}`
`abc`

Then nothing is output.

**Comment by kaldvelski** [comment]
I've come up with a work around that is to wrap the `readFile` call with `replace` to remove the line breaks, i.e.

`{{ tpl (replace "\n" "" (readFile "templates/load.tpl")) .  }}`

**Comment by yxxhero** [comment]
`emplates/load.tpl` @kaldvelski  please show the content of that file. I will try to reproduce this issue.

**Reply by kaldvelski** [reply]
My content wouldn't mean much however it can be reproduced with something as simple as:

`{{ $out := "abc" }}`
`{{ print $out }}`

**Comment by yxxhero** [comment]
@kaldvelski ok. Thanks for your code. I will try to optimize the logic.
